Makefile

```diff
@@ -42,11 +42,15 @@ tinycbor/Makefile crypto/tiny-AES-c/aes.c:
 .PHONY: cbor
 cbor: $(LIBCBOR)
 
-$(LIBCBOR): tinycbor/Makefile
+$(LIBCBOR):
```

Comment:
If you leave that dependency in, it should automatically check out the submodule if it's not there.

Reply:
however it will then only rebuild Cbor if the makefile (which does not happen) changes even if you change architectures, which breaks builds

Reply:
Changing architecture doesn't indicate a code change.  I think if you're going to change architecture you just have to do a `make clean` first.  If the current `make clean` doesn't clean `tinycbor` then that should be fixed, which would then allow this auto-checkout still work *and* an alternate architecture to build correctly after a `make clean`.

Reply:
good point will fix